### ISSUE TITLE
finish-callback not triggering digest cycle

### ISSUE THREAD

**Issue by prime-ralph** [siddii/angular-timer#240]
This happened to me, in case anyone runs into this issue as well:

I made a change on a $scope attribute/variable inside a function I had placed as the 'finish-callback' of a timer, and yet the changes did not happen to be visible upon the view unless you hover upon a button.

As Topher Fangio put it here: [Angular Material App - delayed changes](http://stackoverflow.com/questions/35578330/angular-material-app-delayed-changes), 

> Without digging into the timer code too much, I'm guessing that the finish-callback functionality that it provides may be running at the end of the digest loop, or entirely outside of Angular's digest cycle (which is a bit odd since it's an Angular module).

**Comment by shivenigma** [comment]
+1

**Comment by djoks** [comment]
Having this issue as well.  Spent soo many hours on it. :(  Can you please post one of your "ugly hacks".

**Comment by shivenigma** [comment]
it depends on what you want to achieve, In my case I needed to hide the timer and display a reveal button on that place. I used id and wrote javascript in controller (Yuk!), that worked for me.

In one place I needed to count down until the timer runs out and once the timer finished I need to start a new timer upwards this got bit challenging, I used $scope.$apply and inside it I added new element to another array and used it to create new timer, to hide the old timer I used javascript again in controller ( I vomit twice :P).
